test/kb_Metrics_server_test.py

```diff
@@ -118,6 +187,549 @@ def getImpl(self):
+            self.assertEqual(mnrecord['nice_name'], 'nice_to_have')
+            self.assertEqual(mnrecord['numObj'], 3)
+        for mnrecord in db_mn.find(upd_filter_set2):
+            self.assertEqual(mnrecord['nice_name'], 'nice_to_have_2')
+            self.assertEqual(mnrecord['numObj'], 5)
+
+        # testing updating existing record
+        upd_ret = self.dbi.update_narrative_records(upd_filter_set2,
+                                                    upd_data_set3)
+        self.assertFalse(upd_ret.raw_result.get('upserted'))
+        self.assertTrue(upd_ret.raw_result.get('updatedExisting'))
+        self.assertTrue(upd_ret.raw_result.get('ok'))
+        self.assertEqual(upd_ret.raw_result.get('n'), 1)
+        for mnrecord in db_mn.find(upd_filter_set2):
+            self.assertEqual(mnrecord['last_saved_at'],
+                             datetime.datetime(2019, 1, 24, 19, 35, 48, 1000))
+            self.assertEqual(mnrecord['numObj'], 7)
+            self.assertEqual(mnrecord['object_version'], 2)
+
+    # Uncomment to skip this test
+    # @unittest.skip("skipped test_MetricsMongoDBs_get_user_info")
+    def test_MetricsMongoDBs_get_user_info(self):
+        minTime = 1516307704700
+        maxTime = 1520549345000
+        user_list0 = []
+        user_list = ['shahmaneshb', 'laramyenders', 'allmon', 'boris']
+
+        # testing get_user_info return data
+        users = self.dbi.get_user_info(user_list0, minTime, maxTime)
+        self.assertEqual(len(users), 37)
+        users = self.dbi.get_user_info(user_list, minTime, maxTime)
+        self.assertEqual(len(users), 4)
+        self.assertIn('username', users[0])
+        self.assertIn('email', users[0])
+        self.assertIn('full_name', users[0])
+        self.assertIn('signup_at', users[0])
+        self.assertIn('last_signin_at', users[0])
+        self.assertIn('roles', users[0])
+
+    # Uncomment to skip this test
+    # @unittest.skip("skipped MetricsMongoDBs_aggr_activities_from_wsobjs")
+    def test_MetricsMongoDBs_aggr_activities_from_wsobjs(self):
+        # testing aggr_activities_from_wsobjs return data
+        minTime = 1468540813000
+        maxTime = 1519768865840
+
+        user_acts = self.dbi.aggr_activities_from_wsobjs(minTime, maxTime)
+        self.assertEqual(len(user_acts), 11)
+        self.assertEqual(user_acts[4]['_id']['ws_id'], 29624)
+        self.assertEqual(user_acts[4]['_id']['year_mod'], 2018)
+        self.assertEqual(user_acts[4]['_id']['month_mod'], 2)
+        self.assertEqual(user_acts[4]['_id']['day_mod'], 26)
+        self.assertEqual(user_acts[4]['obj_numModified'], 5)
+
+    # Uncomment to skip this test
+    # @unittest.skip("skipped test_MetricsMongoDBs_list_ws_narratives")
+    def test_MetricsMongoDBs_list_ws_narratives(self):
+        minTime = 1468592344887
+        maxTime = 1519768865840
+
+        # Testing with time limit
+        ws_narrs = self.dbi.list_ws_narratives(minTime, maxTime)
```

Comment:
In this test, you should check minTime and maxTime really applies. For instance, if moddate = 2 in source DB, you should set minTime to 1 and see if expected record returns. Then set minTime to 3 and make sure that record doesn't return.

Reply:
added more tests